File: angular/src/app/models/response/animal-response.ts
import { VilleResponse } from "./ville-response";

export class AnimalResponse {
    public get villeResponse(): VilleResponse {
        return this._villeResponse;
    }
    public set villeResponse(value: VilleResponse) {
        this._villeResponse = value;
    }
    public get espece(): string {
        return this._espece;
    }
    public set espece(value: string) {
        this._espece = value;
    }
    public get age(): number {
        return this._age;
    }
    public set age(value: number) {
        this._age = value;
    }
    public get pseudo(): string {
        return this._pseudo;
    }
    public set pseudo(value: string) {
        this._pseudo = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    constructor(
        private _id: number,
        private _pseudo: string,
        private _age: number,
        private _espece: string,
        private _villeResponse: VilleResponse,
        ) { }

}

File: spring/src/main/java/fr/formation/api/response/AnimalResponse.java
package fr.formation.api.response;

import org.springframework.beans.BeanUtils;

import fr.formation.model.Animal;

public class AnimalResponse {
	private int id;
	private String pseudo;
	private int age;
	private String espece;
	private VilleResponse ville;

	public VilleResponse getVille() {
		return ville;
	}

	public void setVille(VilleResponse ville) {
		this.ville = ville;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getPseudo() {
		return pseudo;
	}

	public void setPseudo(String pseudo) {
		this.pseudo = pseudo;
	}

	public int getAge() {
		return age;
	}

	public void setAge(int age) {
		this.age = age;
	}

	public String getEspece() {
		return espece;
	}

	public void setEspece(String espece) {
		this.espece = espece;
	}

	public static AnimalResponse convert(Animal animal){
		AnimalResponse animalResponse = new AnimalResponse();
		// Ne fonctionne pas avec Ville car != VilleResponse
		BeanUtils.copyProperties(animal, animalResponse);
		// Donc on le rajoute alamano
		animalResponse.setVille(animal.getVille().convert());
		return animalResponse;
	}

}
